annotator-core/src/test/java/edu/ucr/cs/riple/core/CoreTest.java

```diff
@@ -572,4 +573,33 @@ public void nonnullTypeUseAnnotationOnFormalParameterAcknowledgmentTest() {
     // No annotation should be added even though it can reduce the number of errors.
     Assert.assertEquals(coreTestHelper.getLog().getInjectedAnnotations().size(), 0);
   }
+
+  @Test
+  public void fieldAssignNullableNonNullArrayContent() {
+    coreTestHelper
+        .onTarget()
+        .withSourceLines(
+            "Main.java",
+            "package test;",
+            "public class Main {",
+            "   Object[] arr = new Object[1];",
+            "   void run() {",
+            "     arr[0] = null;",
+            "   }",
+            "}")
+        .withExpectedReports(
+            new TReport(
+                new OnField("Main.java", "test.Main", Set.of("arr")),
+                ImmutableList.of(ImmutableList.of(1, 0)),
+                -1))
+        .setPredicate(
+            (expected, found) ->
```

Comment:
I don't adding a custom predicate here is necessary. Is there a reason you added this ?